dotnet/src/Skills/Skills.OpenAPI/Builders/Query/QueryStringBuilder.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Model;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Builders.Query;
+
+/// <summary>
+/// Represents a query string builder for REST API operations.
+/// </summary>
+internal class QueryStringBuilder : IQueryStringBuilder
+{
+    ///<inheritdoc/>
+    public string Build(RestApiOperation operation, IDictionary<string, string> arguments)
+    {
+        var queryStringSegments = new List<string>();
+
+        var queryStringParameters = operation.Parameters.Where(p => p.Location == RestApiOperationParameterLocation.Query);
+
+        foreach (var parameter in queryStringParameters)
+        {
+            //Resolve argument for the parameter.
+            if (!arguments.TryGetValue(parameter.Name, out var argument))
+            {
+                argument = parameter.DefaultValue;
+            }
+
```

Comment:
Not sure if we need this, but per url laws, `argument` could be a collection, i.e.
`&colors=blue,red,green`
...which is why [Microsoft.AspNetCore.Http.IQueryCollection](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.iquerycollection) uses [StringValues](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.primitives.stringvalues) in `KeyValuePair<string, StringValues>`

Reply:
Right, support for various query string serialization styles will be introduced in the next PR:
<img width="808" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/3c418a45-9aa4-4aaa-aee9-9c3d90e0e1ec">